hydra/_internal/utils.py

```diff
@@ -516,8 +516,12 @@ def _locate(path: str) -> Union[type, Callable[..., Any]]:
     else:
         obj = builtins
     for part in parts[n:]:
+        mod = mod + "." + part
         if not hasattr(obj, part):
-            raise ImportError(f"Could not locate '{path}'")
+            try:
+                import_module(mod)
+            except ImportError as e:
+                raise e
```

Comment:
So, if it's an import error, you catch it and raise it.
and if it's not an import error, you just let it fly.
see the problem?

Reply:
You are right. I (wrongly) assumed that `import_module` can only raise `ImportError` (https://github.com/python/cpython/blob/3.8/Lib/importlib/__init__.py#L109)

Reply:
ouch you are right. 

> Can you add an explicit test for a case where the instantiated class has an import error?

You have a test for this already; https://github.com/facebookresearch/hydra/pull/800/files#diff-01be86b771f92654b26fbe57dd65f9e7

Reply:
Even under this wrong assumption it makes no sense to catch and exception and then raise it without doing anything.